retrofit2/src/main/java/com/linecorp/armeria/client/retrofit2/RetrofitClassAwareMeterIdPrefixFunction.java

```diff
@@ -0,0 +1,179 @@
+    private final String serviceTagName;
+
+    /**
+     * Returns a newly created {@link RetrofitClassAwareMeterIdPrefixFunction} with the specified {@code name}
+     * and {@code serviceClass}.
+     */
+    public static MeterIdPrefixFunction of(String name, Class serviceClass) {
+        return builder(name, serviceClass).build();
+    }
+
+    /**
+     * Returns a newly created {@link RetrofitMeterIdPrefixFunctionBuilder} with the specified {@code name}
+     * and {@code serviceClass}.
+     */
+    public static RetrofitMeterIdPrefixFunctionBuilder builder(String name, Class serviceClass) {
+        return new RetrofitMeterIdPrefixFunctionBuilder(requireNonNull(name, "name"))
+                .withServiceClass(serviceClass);
+    }
+
+    RetrofitClassAwareMeterIdPrefixFunction(String name,
+                                            Class serviceClass,
+                                            String serviceTagName) {
+        this.name = name;
+        this.serviceTagName = firstNonNull(serviceTagName, "service");
+        this.serviceName = serviceClass.getSimpleName();
+        this.methodNameToTags = defineTagsForMethods(serviceClass);
+    }
+
+    @Override
+    public MeterIdPrefix activeRequestPrefix(MeterRegistry registry, RequestLog log) {
+        final ImmutableList.Builder<Tag> tagsListBuilder = ImmutableList.builder();
+        buildTags(tagsListBuilder, log);
+        return new MeterIdPrefix(name, tagsListBuilder.build());
+    }
+
+    @Override
+    public MeterIdPrefix apply(MeterRegistry registry, RequestLog log) {
+        final ImmutableList.Builder<Tag> tagListBuilder = ImmutableList.builder();
+        buildTags(tagListBuilder, log);
+        RequestMetricSupport.appendHttpStatusTag(tagListBuilder, log);
+        return new MeterIdPrefix(name, tagListBuilder.build());
+    }
+
+    private void buildTags(ImmutableList.Builder<Tag> tagsBuilder, RequestLog log) {
+        final Invocation invocation = InvocationUtil.getInvocation(log);
+
+        tagsBuilder.add(Tag.of(serviceTagName, serviceName));
+        if (invocation != null) {
+            final String methodName = invocation.method().getName();
+            final List<Tag> methodTags = methodNameToTags.get(methodName);
+            if (methodTags != null) {
+                tagsBuilder.addAll(methodTags);
+                return;
+            }
+        }
+
+        tagsBuilder.add(Tag.of("method", log.method().name()));
+    }
+
+    @VisibleForTesting
+    static Map<String, List<Tag>> defineTagsForMethods(Class serviceClass) {
+        final Builder<String, List<Tag>> methodNameToTags = ImmutableMap.builder();
+
+        final Method[] declaredMethods = serviceClass.getDeclaredMethods();
+        for (final Method clientServiceMethod : declaredMethods) {
+            final Annotation[] declaredAnnotations = clientServiceMethod.getDeclaredAnnotations();
+            if (declaredAnnotations.length == 0) {
+                continue;
+            }
```

Comment:
I wonder if this can be removed without affecting the logic and performance?

Reply:
+1 for removing for simplicity